chain/messagepool/messagepool.go

```diff
@@ -824,7 +824,16 @@ func (mp *MessagePool) VerifyMsgSig(m *types.SignedMessage) error {
 		return nil
 	}
 
-	if err := consensus.AuthenticateMessage(m, m.Message.From); err != nil {
+	mp.curTsLk.RLock()
+	curTs := mp.curTs
+	mp.curTsLk.RUnlock()
+	epoch := curTs.Height() + 1
```

Comment:
I'm a little worried about how this works for null epochs during network upgrade.

Reply:
Actually no Im not